paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -102,14 +139,132 @@ def mark_for_deployment(git_url, deploy_group, service, commit):
+    if not api:
+        # Assume not deployed if we can't reach API
+        return False
+    statuses = []
+    for instance in instances:
+        try:
+            statuses.append(api.service.status_instance(service=service, instance=instance).result())
+        except HTTPError as e:
+            if e.response.status_code == 404:
+                log.warning("Can't get status for instance {0}, service {1} in cluster {2}. "
+                            "This is normally because it is a new service that hasn't been "
+                            "deployed by PaaSTA yet".format(instance, service, cluster))
+                statuses.append(None)
+    results = []
+    for status in statuses:
+        if not status:
+            results.append(False)
+        # if it's a chronos service etc then skip waiting for it to deploy
+        elif not status.marathon:
+            results.append(True)
+        else:
+            results.append(git_sha.startswith(status.git_sha) and
+                           status.marathon.app_count == 1 and
+                           status.marathon.deploy_status == 'Running' and
+                           status.marathon.expected_instance_count == status.marathon.running_instance_count)
+    return all(results)
+
+
+def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
+    cluster_map = get_cluster_instance_map_for_service(soa_dir, service, deploy_group)
+    if not cluster_map:
+        line = "Couldn't find any instances for service {0} in deploy group {1}".format(service, deploy_group)
+        _log(
+            service=service,
+            component='deploy',
+            line=line,
+            level='event'
+        )
+        log.error(line)
+        raise NoInstancesFound
+    for cluster in cluster_map.values():
+        cluster['deployed'] = False
+    try:
+        with Timeout(seconds=timeout):
+            while True:
+                for cluster, instances in cluster_map.items():
+                    if not cluster_map[cluster]['deployed']:
+                        cluster_map[cluster]['deployed'] = are_instances_deployed(cluster=cluster,
+                                                                                  service=service,
+                                                                                  instances=instances['instances'],
+                                                                                  git_sha=git_sha)
+                if all([cluster['deployed'] for cluster in cluster_map.values()]):
+                    break
+                time.sleep(10)
+    except TimeoutError:
+        human_status = ["{0}: {1}".format(cluster, data['deployed']) for cluster, data in cluster_map.items()]
+        line = "\nCurrent deployment status of {0} per cluster:\n".format(deploy_group) + "\n".join(human_status)
+        _log(
+            service=service,
+            component='deploy',
+            line=line,
+            level='event'
```

Comment:
I wouldn't _log this, too noisy to print this every 10 seconds to the scribe log